tests/Doctrine/Tests/ORM/Functional/DatabaseDriverTestCase.php

```diff
@@ -31,7 +31,7 @@ protected function convertToClassMetadata(array $entityTables, array $manyTables
 
         $metadatas = [];
         foreach ($driver->getAllClassNames() as $className) {
-            $class = new ClassMetadataInfo($className);
+            $class = new ClassMetadata($className);
```

Comment:
I first implemented this with `assert`, then was forced to changed this test, which made me think this was a bit risky, hence the change towards phpdoc. It's better anyway, since it makes static analysis able to detect calls to these methods made with `ClassMetadataInfo` rather than `ClassMetadata`.